### PR TITLE
feat(ingest) kafka-connect: prefix server to table name using a flag

### PR DESCRIPTION
**Pull request by arunvasudevan**
## Checklist
- [X ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ X] Links to related issues (if applicable)
- [ X] Docs related to the changes have been added/updated (if applicable)

**Conversation in Slack:**
Based on this conversation here - https://datahubspace.slack.com/archives/CUMUWQU66/p1646256415909719

**Background:**
     When ingesting multiple instances for the same source such as PostgreSQL the urn ingested would be for example: `urn:li:dataset:(urn:li:dataPlatform:postgres,{platform_instance_name}.{db name}.{schema name}.{table name},PROD)`

So when Kafka Connect ingests data from Debezium connector the table name that it builts will have `{db name}.{schema + table name}` 
In order to add the `platform_instance_name` to the table name, 2 things needs to be followed:
 1. Matching its value with `database.server.name` in debezium 
 2. Enabling this flag `prefix_server_to_table_name` to prefix the server name to the table

This will have no impact to existing flows.

### CODE REVIEW

**Review by shirshanka — APPROVED** [review]

**Comment by arunvasudevan** [comment]
@shirshanka Curious why the merge didn't kick in after the Approval?

File: metadata-ingestion/source_docs/kafka-connect.md [hunk 1]
```diff
@@ -58,20 +58,20 @@ sink:
...
+| `env`                          |          | `"PROD"`                   | Environment to use in namespace when constructing URNs.                                                                                                                                                                                                                |
+| `platform_instance_map`        |     |                            | Platform instance mapping to use when constructing URNs. e.g.`platform_instance_map: { "hive": "warehouse" }`                                                                                                                                                          | 
+| `prefix_server_to_table_name`  |          | `False`                    | Whether to prefix Database server name to the table name. When multiple instances are ingested for a single platform then the platform_instance name in the recipe could be matched to the database server name in Kafka connect and there by making the lineage work. |
```
Comment (shirshanka) [inline metadata-ingestion/source_docs/kafka-connect.md:74]: what would the generated urns look like if both platform_instance map is specified and the prefix_server_to_table_name config is set to true?

**Review by shirshanka — CHANGES_REQUESTED** [review]
actually need to understand this PR a bit more...

File: metadata-ingestion/source_docs/kafka-connect.md (hunk 1 above)
Comment (arunvasudevan) [reply metadata-ingestion/source_docs/kafka-connect.md:74]: When `prefix_server_to_table_name` config is set to `True` the urn would look like
`urn:li:dataset:(urn:li:dataPlatform:postgres,{database_server_name}.{db_name}.{schema_name}.{table_name}, PROD)`

For the lineage to work with the data ingested from say Postgres the [database.server.name](https://debezium.io/documentation/reference/stable/connectors/sqlserver.html#sqlserver-topic-names) should be same as `platform_instance` in database ingestion recipe.

When multiple instances for a platform is available `platform_instance_map` cannot be specified for that platform so only `prefix_server_to_table_name` is sufficient.

**Comment by arunvasudevan** [comment]
Closing this PR on for the new Approach PR - https://github.com/datahub-project/datahub/pull/4501 after discussing with @shirshanka

**Pull request closed without merging**